PHYSOR26/common_mesh.i

```diff
@@ -0,0 +1,126 @@
+ref_mesh_file_name=gt.e-s011
+test_mesh_file_name=hex_same_flux_high_err_ma.e-s010
+
+[Mesh]
+  [generated_mesh]
+     type = GeneratedMeshGenerator
+    nx = 20
+    ny = 4
+    nz = 4
+    xmin = 0.0
+    xmax = 100.0
+    ymin = 0.0
+    ymax = 10.0
+    zmin = 0.0
+    zmax = 10.0
+    dim = 3
+  []
+[]
+
+
+[AuxVariables]
+    [adaptive_hierarchy]
+        order = CONSTANT
+        family = MONOMIAL
+    []
+    [element_refinement_step]
+        order = CONSTANT
+        family = MONOMIAL
+    []
+    [test_solution_hierarchy]
+        order = CONSTANT
+        family = MONOMIAL
+    []
+
+    [max_elemental_hierachy]
+        order = CONSTANT
+        family = MONOMIAL
+    []
+    [ref_solution_hierarchy]
+        order = CONSTANT
+        family = MONOMIAL
+    []
+[]
+
+[AuxKernels]
+
+  [adaptive_hierarchy_aux]
+    type=ElementHierarchyAux
+    variable=adaptive_hierarchy
+  []
+
+  [cp_ref_sln_mesh_hierarchy]
+    type= SolutionAux
+    solution = ref_solution_hierarchy_user_object
+    variable = ref_solution_hierarchy
+  []
+
+  [cp_test_sln_mesh_hierarchy]
+    type= SolutionAux
+    solution = test_solution_hierarchy_user_object
+    variable = test_solution_hierarchy
+  []
+
+  [calculate_max_hierarchy_refinement_step]
+    type = ParsedAux
+    variable = max_elemental_hierachy
+    coupled_variables = 'test_solution_hierarchy ref_solution_hierarchy'
+    expression = 'max(test_solution_hierarchy, ref_solution_hierarchy)'
+  []
+
+  [calculate_refinement_step]
+    type = ParsedAux
+    variable = element_refinement_step
+    coupled_variables = 'adaptive_hierarchy max_elemental_hierachy'
+    expression = 'abs(max_elemental_hierachy - adaptive_hierarchy)'
+  []
+[]
+
+[Adaptivity]
+    marker = marker
+    steps = 1
+
+  [Markers]
+    [marker]
+      # for an element  We should be
+      # refining untill the element_refinement_step is zero
+      type = ValueThresholdMarker
+      coarsen = -2
+      refine = 0.7
+      variable = element_refinement_step
+    []
+
+  []
+[]
+
+[UserObjects]
+    [ref_solution_hierarchy_user_object]
+        type = SolutionUserObject
+        mesh = ${ref_mesh_file_name}
+        system_variables = hierarchy
+        timestep = 1
+    [] 
+
+   [test_solution_hierarchy_user_object]
+        type = SolutionUserObject
+        mesh = ${test_mesh_file_name}
+        system_variables = hierarchy
+        timestep = 1
+    []
+[]
+
+[Problem]
+  type = FEProblem
+  solve = false
+[]
+
+[Executioner]
+  type = Transient
+  solve = false
+  dt = 1
+  num_steps = 4
```

Comment:
Can we continue until the remaining refinement steps is 0?

Reply:
We know that a multiphysics calculation can have a variable number of steps based on the physics solution, right?

Reply:
This is probably secondary and less important than cleaning up the rest.  If we have to live with changing this at the command-line for now, that's fine